File: python_backend/src/arch/query/RecipeStepQueries.py
from typing import List
from uuid import UUID

from pony.orm import db_session, select

from arch.data_objects.RecipeStepData import RecipeStepData
from db import RecipeStep


@db_session
def get_steps_for_recipe(recipe_id: UUID) -> List[RecipeStepData]:
    steps = list(select(step for step in RecipeStep if step.recipe.id == recipe_id))
    return [RecipeStepData(
        number=step.number,
        step=step.step
    ) for step in steps]

File: backend/internal/db/queries/recipeStepQueries.go
package queries

import (
	"github.com/georgysavva/scany/pgxscan"
	"github.com/google/uuid"
	"github.com/viddem/vrecipes/backend/internal/db/tables"
)

var getStepsForRecipeQuery = `SELECT recipe_id, number, step FROM recipe_step WHERE recipe_id=$1`

func GetStepsForRecipe(recipeId uuid.UUID) ([]*tables.RecipeStep, error) {
	db := getDb()

	var recipeSteps []*tables.RecipeStep
	err := pgxscan.Select(ctx, db, &recipeSteps, getStepsForRecipeQuery, recipeId)
	return recipeSteps, err
}
